libraries/tarball.rb

```diff
@@ -0,0 +1,141 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: tarball.rb
+#
+# Methods for working with data from a tarball.
+
+require 'rubygems/package'
+require 'zlib'
+require 'fileutils'
+
+module CernerSplunk
+  # Utility class to encapsulate operations on tar.gz files
+  # Mad Props to http://stackoverflow.com/a/19139114/504685
+  class TarBall
```

Comment:
I think I have to just admit defeat in terms of reviewing this class.  Parts of my trouble are probably my limited experience and skills in Ruby and parts are probably the overall complexity of the code in general.  Anyhow, I just want to note that I'm not really reviewing this specific piece.  I suspect others at least more versed in Ruby will have looked over it.